sycl/doc/extensions/supported/sycl_ext_oneapi_bfloat16.asciidoc

```diff
@@ -0,0 +1,323 @@
+= sycl_ext_oneapi_bfloat16
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+// This is necessary for asciidoc, but not for asciidoctor
+:cpp: C++
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 5.
+
+== Status
+
+This extension is implemented and fully supported by DPC++.
+[NOTE]
+====
+The DPC++ compiler has the following limitation when using this extension
+in conjunction with ahead-of-time (AOT) compilation with the `-fsycl-targets`
+compiler option.  When doing AOT compilation for an Intel GPU device via
+`-fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device ..."`,
+the compiler chooses either fallback or native support for bfloat16 according
+to the device(s) specified in `...`.  Native support is used only if all of
+these devices have native bfloat16 support.  As a result, AOT compiling for
+multiple Intel GPU devices could result in the lower performance fallback
+support even when running on a GPU that has native support.  Therefore, the
+recommendation is to use AOT only when all Intel GPU devices have the same
+type of bfloat16 support (all native support or all fallback support).
+
+There is a similar limitation when AOT compiling for one Intel GPU device and
+running on a different Intel GPU device. In this case, the compiler chooses
+either fallback or native bfloat16 support according to the device(s) specified
+on the command line. If the fallback library was chosen at AOT compilation
+time, then the binary will run on all Intel GPU devices but you will not
+get the performance benefit of native support even when running on a new
+Intel GPU that has native support. If however, the native
+bfloat16 library had been chosen at AOT compilation time then the binary
```

Comment:
Minor nit on grammar:

```suggestion
bfloat16 library was chosen at AOT compilation time then the binary
```

Reply:
In my opinion the grammar is OK. The first two sentences describe the compilation step. The next two sentences are set in the future and refer to the compilation step as in the past. That's why the third sentence says "was chosen at AOT compilation time". And consistent with that, the fourth sentence uses "had been". That third sentence could also have used "had been" but I think "was" there has the same meaning.

Reply:
ok